src/components/PopupWithForm.js

```diff
@@ -1,36 +1,46 @@
-import PopUp from "./PopUp.js";
+import PopUp from "./Popup.js";
 
 export default class PopUpWithForm extends PopUp {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputEls = Array.from(
-      this._popupForm.querySelectorAll(".modal__input")
-    );
+      this._popupForm.querySelectorAll(".modal__input"));
+    this._popupButton = this._popupForm.querySelector(".modal__button");
   }
+  
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
   _getInputValues() {
-    const formValues = [];
+    const formValues = {};
     this._inputEls.forEach((element) => {
       formValues[element.name] = element.value;
     });
 
     return formValues;
   }
 
+  submitLoading(isLoading) {
+    if (isLoading) {
+      this._popupButton.textContent = "Saving...";
+    } else {
+      this._popupButton.textContent = "Save";
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`